Mathlib/Algebra/Homology/DerivedCategory/HomologySequence.lean

```diff
@@ -0,0 +1,101 @@
+/-
+Copyright (c) 2024 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+
+import Mathlib.Algebra.Homology.DerivedCategory.Basic
+
+/-!
+# The homology sequence
+
+In this file, we construct `homologyFunctor C n : DerivedCategory C ⥤ C` for all `n : ℤ`,
+show that they are homological functors which form a shift sequence, and construct
+the long exact homology sequences associated to distinguished triangles in the
+derived category.
+
+-/
+
+universe w v u
+
+open CategoryTheory Pretriangulated
+
+variable (C : Type u) [Category.{v} C] [Abelian C] [HasDerivedCategory.{w} C]
+
+namespace DerivedCategory
+
+/-- The homology functor `DerivedCategory C ⥤ C` in degree `n : ℤ`. -/
+noncomputable def homologyFunctor (n : ℤ) : DerivedCategory C ⥤ C :=
+  HomologicalComplexUpToQuasiIso.homologyFunctor C (ComplexShape.up ℤ) n
+
+/-- The homology functor on the derived category is induced by the homology
+functor on the category of cochain complexes. -/
+noncomputable def homologyFunctorFactors (n : ℤ) : Q ⋙ homologyFunctor C n ≅
+    HomologicalComplex.homologyFunctor _ _ n :=
+  HomologicalComplexUpToQuasiIso.homologyFunctorFactors C (ComplexShape.up ℤ) n
+
+/-- The homology functor on the derived category is induced by the homology
+functor on the homotopy category of cochain complexes. -/
+noncomputable def homologyFunctorFactorsh (n : ℤ) : Qh ⋙ homologyFunctor C n ≅
+    HomotopyCategory.homologyFunctor _ _ n :=
+  HomologicalComplexUpToQuasiIso.homologyFunctorFactorsh C (ComplexShape.up ℤ) n
+
+instance (n : ℤ) : (homologyFunctor C n).IsHomological :=
+  Functor.isHomological_of_localization Qh
+    (homologyFunctor C n) _ (homologyFunctorFactorsh C n)
+
+/-- The functors `homologyFunctor C n : DerivedCategory C ⥤ C` for all `n : ℤ` are part
+of a "shift sequence", i.e. they satisfy compatiblities with shifts. -/
+noncomputable instance : (homologyFunctor C 0).ShiftSequence ℤ :=
+  Functor.ShiftSequence.induced (homologyFunctorFactorsh C 0) ℤ
+    (homologyFunctor C) (homologyFunctorFactorsh C)
+
+variable {C}
+
+namespace HomologySequence
+
+variable (T : Triangle (DerivedCategory C)) (hT : T ∈ distTriang _)
+  (n₀ n₁ : ℤ) (h : n₀ + 1 = n₁)
+
+/-- The connecting homomorphism on the homology sequence attached to a distinguished
+triangle in the derived category. -/
+noncomputable def δ : (homologyFunctor C n₀).obj T.obj₃ ⟶ (homologyFunctor C n₁).obj T.obj₁ :=
+  (homologyFunctor C 0).shiftMap T.mor₃ n₀ n₁ (by rw [add_comm 1, h])
+
+lemma comp_δ : (homologyFunctor C n₀).map T.mor₂ ≫ δ T n₀ n₁ h = 0 :=
```

Comment:
```suggestion
@[reassoc]
lemma comp_δ : (homologyFunctor C n₀).map T.mor₂ ≫ δ T n₀ n₁ h = 0 :=
```
Or
```suggestion
@[reassoc (attr := simp)]
lemma comp_δ : (homologyFunctor C n₀).map T.mor₂ ≫ δ T n₀ n₁ h = 0 :=
```

Reply:
Thanks! I have just added the reassoc/simp attribute.